grafana/grafana
# [X] Correlations: Allow creating correlations for provisioned data sources

Fixes #71110

This is to allow creating correlations for provisioned data sources. Please check  #71110 for more details and design doc.

Key changes:
* Clean-up logic was adjusted to ensure correlations are removed when a data source is removed, taking into account a scenario when a data source is re-created during provisioning and no data loss occurs
* Previously, correlations attached to a read-only data source were marked as read-only. Now, only provisioned correlations are read-only 

**Special notes for your reviewer:**

<details><summary>How to test it?</summary>

You can use your own/current setup or follow manual test scenario I've created: https://gist.github.com/ifrost/35b7d7424a625d9cd8fc8b845fbd002a

Try out different things:
* Adding new correlations to different data sources (provisioned and not provisioned)
* Removing manually added correlations
* Add `deleteDatasources` section in provisioning config with different datasources that are being added later; then try actually removing some of the data sources
* Adding/Editing/Removing correlations inside provisioning file
* Restart Grafana with the same config

After each step, you should see correlations belonging to existing data sources. Manual entries should not be removed during provisioning. Check the sqlite db directly, not only frontend (we filter out correlations for data sources that do not exist)

</details>

Please check that:
